src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/UserService.kt

```diff
@@ -329,7 +329,7 @@ class UserService(
     }
 
     if (forService == ServiceName.approvedPremises) {
-      val apArea = cas1ApAreaMappingService.determineApArea(user.probationRegion, staffDetail.teamCodes(), staffDetail.username)
+      val apArea = cas1ApAreaMappingService.determineApArea(user.probationRegion, staffDetail.teamCodes(), staffDetail.username!!)
```

Comment:
It's a shame that username is now nullable to Accommodate the get by staff code API. We obviously know this isn't null when we're using the 'get by username' endpoint - can you think of anyway to 'bake this in' to the API client response or is it going to be too awkward? Maybe worth having a chat about this

it's also worth noting that we don't appear to use the username field when retrieving staff via code?

Reply:
Following on from our discussion:

* whilst staff legitimately may not have a username defined, when retrieving staff details via `getStaffDetail(username:String)` we know that the returned user (if any) _must_ have a username
* our first idea was to remove the username? field from the API response to force the calling code to use whatever username was passed into the client function, but this field was used in API responses, so could break something in the UI
* the agreed approach was to instead avoid using staffDetail.username in calling code, and instead use the non-null username value used when calling `getStaffDetail(username:String)`. This may require adding it as a function argument whereever the result of this API call is used.

Reply:
looks good!